### ISSUE TITLE
Windows MSI installer

### ISSUE THREAD

**Issue by tomswartz07** [owncloud/client#860]
Hi all,

It is possible to install OwnCloud remotely, via a combination of PSExec and Network Shares (as an example, for installing:
psexec \machinename \path\to\owncloud-1.2.5-setup.exe /S)

I, as I'm sure many other corporate and system admins, would appreciate a pre-built MSI installer for Windows.

While tools exist to create an MSI from an EXE, and there are some (very questionable) pre-made ownCloud MSI's available, having an 'OwnCloud Blessed' MSI would greatly help speed up deployment.

Thank you 
-Tom Swartz

**Comment by crysman** [comment]
Waiting for 1.5 than and hoping for the MSI support added, it will be very valuable for company GPO domain deployment, which is what I need right now. Wish you good luck and thank you for improving and developing ownCloud!

**Comment by danimo** [comment]
@crysman 1.5 is long released. @dschmidt expressed some interest in working on it during the next hackathon in Berlin. Generally this currently is pushed back in the backlog, unless someone puts effort into it, which many Windows admins probably would appreciate.

**Comment by tomswartz07** [comment]
@crysman I have an 'experimental' branch that will create an MSI via the program WiX.

You can grab the code [here](https://github.com/pennmanor/wix-wrapper) for ownCloud client 1.8.1

I say 'experimental' because it's not officially part of the client codebase. In practice, It's fairly rock solid and we use it here to deploy the client to over 1k client machines via Active Directory Group Policy.

**Comment by crysman** [comment]
@tomswartz07 :+1: seems to be what I am looking for - is it going to be maintained for further versions? I mean if you are planning to maintain it. Thank you.

**Comment by tomswartz07** [comment]
@crysman you betcha. I usually have that repo updated within a few hours of the official release.

**Comment by crysman** [comment]
According to this comment
https://github.com/owncloud/client/issues/3949#issuecomment-148157711

I am requesting here the option to __install ownCloud clients without shortcuts__ being created on desktops.

PS: it all started here, actually https://github.com/pennmanor/wix-wrapper/issues/6

**Comment by AnrDaemon** [comment]
*sigh*

**Comment by fcturner** [comment]
Like I said on #3011 (because I think you should all know this):
>I see that this issue is almost a year old now. @danimo You tell us it depends on #860. Well, that issue is being pushed to the future ever since 2013, and only changes milestones. Exactly the same as with this issue.. If I would constantly postpone my deadlines at work, my boss wouldn't be very happy :smile:

**Comment by michaelstingl** [comment]
@MTRichards @danimo What is the status here? /cc @MorrisJobke 

00004916

**Comment by MTRichards** [comment]
Currently on the backlog, but not planned for 2.2. We can look at it for 2.3. 

This is a long standing request, and we will get to it.

**Comment by MTRichards** [comment]
@cmonteroluque FYI.

**Comment by cmonteroluque** [comment]
@MTRichards yeah

**Comment by fcturner** [comment]
:+1:

**Comment by AnrDaemon** [comment]
Guys, if you just want to be subscribed for notifications, there's a button at the right >.<

**Comment by DraconPern** [comment]
I have experience making MSI's so I can probably work on this.  There's a also discussion at my work about using owncloud and if there is an MSI, it'll be very easy for admins to install the client on all the workstations.  So, IT admins, what features do you want to see in the MSI?

**Comment by tomswartz07** [comment]
Hey there @DraconPern: 

I currently have a workaround using WiX Toolkit. The code 'wraps' the current `.exe` in an `.msi` and can be deployed using any number of admin tools.
We have used this method to deploy the program to over 2k machines in the past few years.
The code is [here](https://github.com/pennmanor/wix-wrapper) to use that.

As far as an 'official' MSI installer, I would love to see:
- Fully configurable install options using `.mst` transform files
- Silent install options
- Options to specify alternate 'exclude.lst' entries during install (ideally as part of the transform)

Cheers

**Comment by danimo** [comment]
@DraconPern The only real requirement is to use WiX to build the installer. Also, we need to somehow make sure that we can use Transifex to translate it (probably needs conversion from the translations that WiX accepts). I can help with the scripts, but I would first like to have a minimal framework before deciding where to go in terms of installer features.

**Comment by AnrDaemon** [comment]
@DraconPern, only configuration coming to mind is configurable default sync directory.
And probably configurable list of default ignored files.

**Comment by AnrDaemon** [comment]
A cherry on top: https://packetstormsecurity.com/files/134694/jrsoft-dllhijack.txt

**Comment by danimo** [comment]
@AnrDaemon We're using NSIS, not InnoSetup. NSIS also suffered from the same issue, but was fixed since.

**Comment by guruz** [comment]
-> https://github.com/owncloud/client/issues/1419
